erpnext/shopping_cart/cart.py

```diff
@@ -48,6 +48,67 @@ def get_cart_quotation(doc=None):
 		"shipping_rules": get_applicable_shipping_rules(party)
 	}
 
+@frappe.whitelist()
+def apply_coupon_code(applyed_code):
```

Comment:
Please also write generic method for the same so that coupon code can be used inside desk, pos or shopping cart. If you write generic one, you don't need to write this method, `get_cart_quotation` can be used for the same then.

Check the following for more reference
1. Frappe call for getting the item details. Also includes auto apply of the pricing rule.
https://github.com/frappe/erpnext/blob/59c915f5998533a3bb14407caafc9b11f1f798a9/erpnext/public/js/controllers/transaction.js#L315
2. Python method for applying the pricing rule
https://github.com/frappe/erpnext/blob/59c915f5998533a3bb14407caafc9b11f1f798a9/erpnext/accounts/doctype/pricing_rule/pricing_rule.py#L79